src/main/java/org/myrobotlab/service/TesseractOcr.java

```diff
@@ -35,70 +44,60 @@
 public class TesseractOcr extends Service {
 
   private static final long serialVersionUID = 1L;
-  transient private TessBaseAPI api;
-
+  transient private TessBaseAPI api = null;
+  // define which language model to load
+  String lang = "eng";
   public final static Logger log = LoggerFactory.getLogger(TesseractOcr.class);
 
-  public static void main(String[] args) {
-    LoggingFactory.init(Level.INFO);
-
-    try {
-
-      TesseractOcr tesseract = (TesseractOcr) Runtime.start("tesseract", "TesseractOcr");
-      // String found = tesseract.ocr("phototest.jpg");
-      // String found = tesseract.ocr("30.speed.JPG");
-      String found = tesseract.ocr("traffic.sign.jpg");
+  public TesseractOcr(String n, String id) {
+    super(n, id);
+  }
 
-      // String found = tesseract.ocr("test.jpg");
-      // String found = tesseract.ocr("test.tif");
-      log.info("found {}", found);
-      Runtime.start("gui", "SwingGui");
+  @Override
+  synchronized public void startService() {
+    super.startService();
+    initModel(lang);
+  }
 
-    } catch (Exception e) {
-      Logging.logError(e);
-    }
+  @Override
+  synchronized public void stopService() {
+    super.stopService();
+    // stop the tesseract api.
+    // TODO: additional memory/object cleanup?
+    api.End();
   }
 
-  public TesseractOcr(String n, String id) {
-    super(n, id);
+  private void initModel(String lang) {
+    api = new TessBaseAPI();
+    String tessData = "resource/TesseractOcr";
+    // load the models
+    if (api.Init(tessData, lang) != 0) {
+      log.warn("Unable to load tesseract model in {} with language {}", tessData, lang);
+      // TODO: what's the lifecycle if a service fails to start?  maybe we should throw?
+    }
   }
 
   public String ocr(BufferedImage image) throws IOException {
-    // A class called PIXConversions will convert directly
-    // but I'm afraid to import the library in that it might clash with the
-    // version
-    // javacv is expecting.. so we are going to do it the "file" way :P
-    File temp = File.createTempFile("tesseract", ".jpg");
+    // The tesseract api has some issues reading the image directly as a byte array.
+    // so for now... until that changes, we'll write a temp file to be ocr'd and 
+    // TODO: where is this.. and i think we need to randomize the name  of it
+    // or synchronize this method.
+    File temp = File.createTempFile("tesseract", ".png");

```

Comment:
Perhaps manage your own temp file in the resource or data dir of TesseractOcr ?
I might be wrong, but I think your file is only guaranteed to be removed with the jvm exits, so this becomes a burden of file handles for long running processes

Reply:
good idea.  done.